examples/example-xds/src/main/java/io/grpc/examples/helloworldxds/HelloWorldServerXds.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2015 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.helloworldxds;
+
+import io.grpc.Server;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.stub.StreamObserver;
+import io.grpc.xds.internal.sds.XdsServerBuilder;
+import java.io.IOException;
+import java.util.logging.Logger;
+
+/**
+ * An xDS-managed Server for the {@code Greeter} service.
+ */
+public class HelloWorldServerXds {
+  private static final Logger logger = Logger.getLogger(HelloWorldServerXds.class.getName());
+  private final int port;
+  private final boolean useXdsCreds;
+  private final String hostName;
+  private Server server;
+
+  public HelloWorldServerXds(int port, String hostName, boolean useXdsCreds) {
+    this.port = port;
+    this.hostName = hostName;
+    this.useXdsCreds = useXdsCreds;
+  }
+
+  private void start() throws IOException {
+    XdsServerBuilder builder = XdsServerBuilder.forPort(port).addService(new HostnameGreeter(hostName));
+    if (useXdsCreds) {
+      builder = builder.useXdsSecurityWithPlaintextFallback();
+    }
+    server = builder.build().start();
+    logger.info("Server started, listening on " + port);
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread() {
+              @Override
+              public void run() {
+                // Use stderr here since the logger may have been reset by its JVM shutdown hook.
+                System.err.println("*** shutting down gRPC server since JVM is shutting down");
+                HelloWorldServerXds.this.stop();
+                System.err.println("*** server shut down");
+              }
+            });
+  }
+
+  private void stop() {
+    if (server != null) {
+      server.shutdown();
```

Comment:
`server.shutdown().awaitTermination(30, TimeUnit.SECONDS);` Try keep as close as the existing code in the existing hello world example.